src/sdkWrapper.ts

```diff
@@ -333,6 +335,41 @@ export class UnsClientWrapper {
         return discloseDemandCertification.data;
     }
 
+    /**
+     * Create Unik voucher
+     * @param explicitValue
+     * @param type
+     * @param coupon
+     * @param paymentProof
+     */
+    public async createUnikVoucher(
+        explicitValue: string,
+        type: DIDType,
+        coupon?: string,
+        paymentProof?: string,
+    ): Promise<string> {
+        try {
+            const createVoucherResponse: Response<UnikVoucherResult> = await this.unsClient.vouchers.create({
+                explicitValue,
+                didType: DIDHelpers.fromLabel(type),
+                coupon,
+                paymentProof,
+            });
+
+            if (createVoucherResponse.error) {
+                throw createVoucherResponse.error;
+            }
+
+            const voucher: string | undefined = createVoucherResponse.data?.unikVoucher;
+            if (!voucher) {
+                throw new Error("Created UNIK voucher is invalid");
+            }
```

Comment:
Useless, data can be undefined by design but will never be. If data is undefined it's because erreor is defined, and it's check some lines before.

You can remove the try/catch and throw directly createVoucherResponse.error